packages/core/src/fhir-deduplication/__tests__/group-same-diag-reports.test.ts

```diff
@@ -12,31 +12,105 @@ let diagReportId: string;
 let diagReportId2: string;
 let diagReport: DiagnosticReport;
 let diagReport2: DiagnosticReport;
+let practId: string;
+let practId2: string;
+let practRef: Reference<Practitioner>;
+let practRef2: Reference<Practitioner>;
 
 beforeEach(() => {
+  practId = faker.string.uuid();
+  practId2 = faker.string.uuid();
+
+  practRef = { reference: `Practitioner/${practId}` };
+  practRef2 = { reference: `Practitioner/${practId2}` };
+
   diagReportId = faker.string.uuid();
   diagReportId2 = faker.string.uuid();
   diagReport = makeDiagnosticReport({ id: diagReportId });
   diagReport2 = makeDiagnosticReport({ id: diagReportId2 });
 });
 
 describe("groupSameDiagnosticReports", () => {
-  it("correctly groups duplicate diagReports based on effectiveDateTime and data presence", () => {
+  it("groups diagReports with the same result", () => {
+    diagReport.result = resultExample;
+    diagReport2.result = resultExample;
+
+    const { diagReportsMap } = groupSameDiagnosticReports([diagReport, diagReport2]);
+    expect(diagReportsMap.size).toBe(1);
+  });
+
+  it("groups diagReports with the same result even if one is missing date", () => {
+    diagReport.result = resultExample;
+    diagReport2.result = resultExample;
+    diagReport.effectiveDateTime = dateTime.start;
+
+    const { diagReportsMap } = groupSameDiagnosticReports([diagReport, diagReport2]);
+    expect(diagReportsMap.size).toBe(1);
+  });
+
+  it("groups diagReports with the same presentedForm", () => {
+    diagReport.result = resultExample;
+    diagReport2.result = resultExample;
+
+    const { diagReportsMap } = groupSameDiagnosticReports([diagReport, diagReport2]);
+    expect(diagReportsMap.size).toBe(1);
+  });
+
+  it("groups diagReports with the same effectiveDateTime and performer ref", () => {
     diagReport.effectiveDateTime = dateTime.start;
     diagReport2.effectiveDateTime = dateTime.start;
+    diagReport.performer = [practRef];
+    diagReport2.performer = [practRef];
+
     diagReport.result = resultExample;
     diagReport2.presentedForm = presentedFormExample;
 
     const { diagReportsMap } = groupSameDiagnosticReports([diagReport, diagReport2]);
     expect(diagReportsMap.size).toBe(1);
   });
 
+  it("does not group diagReports with the same effectiveDateTime, if one of them has a performer and the other one does not", () => {
+    diagReport.effectiveDateTime = dateTime.start;
+    diagReport2.effectiveDateTime = dateTime.start;
+    diagReport.performer = [practRef];
+
+    diagReport.result = resultExample;
+    diagReport2.presentedForm = presentedFormExample;
+
+    const { diagReportsMap } = groupSameDiagnosticReports([diagReport, diagReport2]);
+    expect(diagReportsMap.size).toBe(2);
+  });
```

Comment:
It would be good to validate the array/map has the actual elements we expect, something like

```ts
expect(diagReportsMap.map(d => d.id).toEqual(
  expect.arrayContaining([diagReport.id, diagReport2.id])
);
```

On this test and the ones below

Reply:
done